### PR TITLE
Feature/init dashboards metrics

### PR DESCRIPTION
**Pull request by lennarkivimae**

### CODE REVIEW

File: server/initializer/initializer.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+	for _, dashboard := range dashboards {
+		payloadData := createDashboardPayload(dashboard.Uid, dashboard.Title, api.GrafanaUrl)
+
+		payload.ProcessPayload(cache, payloadData, logger)
+	}
+}
+
+func createDashboardPayload(dashboardUid string, dashboardName string, grafanaUrl string) payload.Payload {
+	id := uuid.New()
```
Comment (atsu85) [inline server/initializer/initializer.go]: perhaps inline this variable, as it is defined quite far from its only usage?

File: server/initializer/initializer.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	}
+
+	for _, dashboard := range dashboards {
+		payloadData := createDashboardPayload(dashboard.Uid, dashboard.Title, api.GrafanaUrl)
+
+		payload.ProcessPayload(cache, payloadData, logger)
+	}
+}
+
+func createDashboardPayload(dashboardUid string, dashboardName string, grafanaUrl string) payload.Payload {
```
Comment (atsu85) [inline server/initializer/initializer.go:25]: misleading name, as it creates panel, not dashboard?

File: server/initializer/initializer.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import (
+	"github.com/MacroPower/macropower-analytics-panel/server/cacher"
+	"github.com/MacroPower/macropower-analytics-panel/server/payload"
+	"github.com/MacroPower/macropower-analytics-panel/server/worker"
+	"github.com/go-kit/kit/log"
+	"github.com/google/uuid"
+	"time"
+)
+
+func InitializeMetricsForDashboards(api worker.Client, logger log.Logger, cache *cacher.Cacher) {
```
Comment (atsu85) [inline server/initializer/initializer.go:12]: misleading name, as it doesn't initialize metrics, but instead adds a panel that starts sending metrics?

File: server/worker/api.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return body, nil
+}
+
+func (api *Client) prepare(method string, endpoint string, payload []byte) (*http.Request, error) {
```
Comment (atsu85) [inline server/worker/api.go]: perhaps rename to `prepareRequest`, to make its usage readable?

**Review by atsu85 — CHANGES_REQUESTED** [review]

File: server/worker/api.go [hunk 5]
```diff
@@ -81,3 +124,18 @@ func (api *Client) Post(endpoint string, payload []byte) ([]byte, error) {
...
+	bearer := "Bearer " + api.Token
+
+	req, err := http.NewRequest(method, api.GrafanaUrl+endpoint, bytes.NewBuffer(payload))
+	if err != nil {
+		return nil, errors.New("Failed to create new HTTP Request")
+	}
+
+	req.Header.Add("Authorization", bearer)
+	req.Header.Add("Content-Type", "application/json")
```
Comment (tiithansen) [inline server/worker/api.go]: Remove, keep this as universal for any HTTP method. GET requests do not have body so `Content-Type` is not needed here.

File: server/worker/api.go (hunk 4 above)
Comment (tiithansen) [reply server/worker/api.go]: I agree, move payload adding logic out from here also, essentially its helper to set some headers (including auth)

File: server/initializer/initializer.go (hunk 3 above)
Comment (tiithansen) [reply server/initializer/initializer.go:12]: It emulates visits to a dashboards to init metrics with 0 values. Username is used to compare if its emulated visit or not. If its emulated then session count (metric value) is not increased.

File: server/initializer/initializer.go (hunk 2 above)
Comment (tiithansen) [reply server/initializer/initializer.go:25]: It creates payload to emulate dashboard visit.

**Review by tiithansen — APPROVED** [review]

**Pull request merged**